### PR TITLE
(docs) Revise filter function docs.

### PR DESCRIPTION
**Pull request by gguillotte**
Update to follow the `map` function revisions.

### CODE REVIEW

File: lib/puppet/functions/filter.rb [hunk 1]
```diff
@@ -1,35 +1,71 @@
-# Applies a parameterized block to each element in a sequence of entries from the first
-# argument and returns an array or hash (same type as left operand for array/hash, and array for
-# other enumerable types) with the entries for which the block evaluates to `true`.
+# Applies a [lambda](http://docs.puppetlabs.com/puppet/latest/reference/lang_lambdas.html)
+# to every value in a data structure and returns an array or hash containing any entries
```
Comment (nfagerlund) [inline lib/puppet/functions/filter.rb]: "containing any elements"

File: lib/puppet/functions/filter.rb [hunk 2]
```diff
@@ -1,35 +1,71 @@
...
+# the lambda evaluates to `true`.
```
Comment (nfagerlund) [inline lib/puppet/functions/filter.rb]: "for which the lambda…"

**Comment by nfagerlund** [comment]
Other than those two proofreads, this looks :+1: to me.

**Comment by gguillotte** [comment]
Amended the previous commit and confirmed all changes are present.

**Comment by nfagerlund** [comment]
I'm :+1: on this!

**Pull request merged**